src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx

```diff
@@ -0,0 +1,252 @@
+      setValidationError(balanceValidationError || ValidationErrors.INVALID_SIGNATURE);
+    } else {
+      if (unsignedTx) {
+        onResult([signature], [unsignedTx]);
+      }
+    }
+  };
+
+  const walletName = session?.peer.metadata.name || t('operation.walletConnect.defaultWalletName');
+
+  return (
+    <div className="flex flex-col items-center p-4 gap-y-2.5 w-[440px] rounded-b-lg">
+      <SmallTitleText>
+        {t('operation.walletConnect.signTitle', {
+          walletName,
+        })}
+      </SmallTitleText>
+
+      <Countdown countdown={countdown} />
+
+      <div className="relative">
+        <video className="object-contain h-[240px]" autoPlay loop>
+          <source src={wallet_connect_confirm_webm} type="video/webm" />
+          <source src={wallet_connect_confirm} type="video/mp4" />
+        </video>
+
+        {validationError === ValidationErrors.EXPIRED && (
+          <>
+            <div className="absolute top-0 left-0 right-0 bottom-0 bg-white opacity-70" />
+            <div className="absolute top-0 left-0 right-0 bottom-0 gap-4 flex flex-col items-center justify-center">
+              <FootnoteText>{t('operation.walletConnect.expiredDescription')}</FootnoteText>
+              <Button size="sm" onClick={onGoBack}>
+                {t('operation.walletConnect.tryAgainButton')}
+              </Button>
+            </div>
+          </>
+        )}
+      </div>
+
+      <div className="flex w-full justify-between mt-5">
+        <Button variant="text" onClick={onGoBack}>
+          {t('operation.goBackButton')}
+        </Button>
+      </div>
+
+      <ConfirmModal
+        panelClass="w-[300px]"
+        isOpen={isReconnectModalOpen}
+        confirmText={t('operation.walletConnect.reconnect.confirmButton')}
+        cancelText={t('operation.walletConnect.reconnect.cancelButton')}
+        onClose={onGoBack}
+        onConfirm={handleReconnect}
+      >
+        <SmallTitleText align="center">
+          {t('operation.walletConnect.reconnect.title', {
+            walletName,
+          })}
+        </SmallTitleText>
+        <FootnoteText className="mt-2 text-text-tertiary" align="center">
+          {t('operation.walletConnect.reconnect.description')}
+        </FootnoteText>
+      </ConfirmModal>
+
+      <StatusModal
+        title={t('operation.walletConnect.reconnect.reconnecting')}
+        isOpen={isReconnectingModalOpen}
+        content={<Animation animation={Animations.loading} loop />}
+        onClose={onGoBack}
+      />
+
+      <StatusModal
+        title={t('operation.walletConnect.reconnect.connected')}
+        isOpen={isConnectedModalOpen}
+        content={<Animation animation={Animations.success} />}
+        onClose={() => setIsConnectedModalOpen(false)}
+      />
+
+      <StatusModal
+        title={t('operation.walletConnect.rejected')}
+        content={<Animation animation={Animations.error} />}
+        isOpen={isRejectedStatusOpen}
+        onClose={onGoBack}
+      />
```

Comment:
Using same Animation object (Animations.success) repeatedly leads to memory leak cause lottie mutates animationData object for some reason. To prevent that you need to deep clone animationData object every time you use an animation. That's why OperationResult component has this effect.
```
useEffect(() => {
    if (isOpen) {
      // using same animation repeatedly without deep clone lead to memory leak
      // https://github.com/airbnb/lottie-web/issues/1159
      setAnimation(JSON.parse(JSON.stringify(Animations[variant])));
    }
  }, [isOpen, variant]);
```

I suggest we move that effect inside Animation component itself so we can forget about that bug and also in that case we can remove OperationResult component and just use StatusModal everywhere.

Reply:
Also I think its better to have one StatusModal component and just pass different props to it like in Staking or Transfer Submit step